packages/template-retail-react-app/app/components/social-login/index.jsx

```diff
@@ -38,44 +38,61 @@ const IDP_CONFIG = {
  */
 const SocialLogin = ({idps}) => {
     const {formatMessage} = useIntl()
+    let missingConfig = false
```

Comment:
You might look at implementing a more reacty way of doing something like this. Using hooks..

e.g.

```
useEffect(() => {
   // Filter you list of idps only showing those that aren't valid,
   // If that list of invalid idps is greater than 0 print out the warning.
}, [idps])
```

Then in your render you would do something like the converse, which would be to filter your idps to only the valid idps and iterate over that. 

If you don't think you understand what I'm saying, react out and I can walk you through it. 

With this kind of solution you won't get warnings for every render.. and you don't need a flag that you flip.